wagtailmedia/static/wagtailmedia/js/WagtailMediaBlock.js

```diff
@@ -0,0 +1,206 @@
+        let left = 0;
+
+        if(direction == 'left'){
+            left = rect.width + 50;
+            top  = rect.top - containerRect.top + (rect.height / 2);
+        }
+        else if (direction == 'top-left'){
+            top = rect.top - containerRect.top + rect.height;
+        }
+
+        this.setState({
+            showTooltipAt: {
+                container: container,
+                top: top,
+                left: left,
+                width: rect.width,
+                height: rect.height,
+                direction: direction,
+            }
+        });
+    }
+
+    closeTooltip(e) {
+        if(e.target.classList){
+            if(e.target.classList.contains("Tooltip__button")){
+                return; // Don't setState if the "Delete" button was clicked
+            }
+        }
+        this.setState({ showTooltipAt: null });
+    }
+
+    /**
+     * Returns either a tooltip "portal" element or null
+     */
+    renderTooltip(data) {
+        const { showTooltipAt } = this.state;
+        const { blockProps } = this.props;
+        const { entity, onRemoveEntity } = blockProps;
+
+        // No tooltip coords exist, don't show one
+        if(!showTooltipAt) return null;
+
+        let options = []
+        if(data.autoplay) options.push("Autoplay");
+        if(data.mute) options.push("Mute");
+        if(data.loop) options.push("Loop");
+        const options_str = options.length ? options.join(", ") : "";
+
+        return ReactDOM.createPortal(React.createElement('div', null,
+            React.createElement('div',
+                {
+                    style: {
+                        top: showTooltipAt.top,
+                        left: showTooltipAt.left
+                    },
+                    class: "Tooltip Tooltip--"+showTooltipAt.direction,
+                    role:  "tooltip"
+                },
+                React.createElement('div', { style: { maxWidth: showTooltipAt.width } }, [
+                    React.createElement('p', {
+                        class: "ImageBlock__alt"
+                    }, data.type.toUpperCase()+": "+data.title),
+                    React.createElement('p', { class: "ImageBlock__alt" }, options_str),
+                    React.createElement('button', {
+                        class: "button button-secondary no Tooltip__button",
+                        onClick: onRemoveEntity
+                    }, "Delete")
+                ])
+            )
+        ), showTooltipAt.container);
+    }
+
+    render() {
+        const { blockProps } = this.props;
+        const { entity } = blockProps;
+        const data = entity.getData();
+
+        let icon;
+        if(data.type == 'video'){
+            icon = React.createElement('span',  { class:"icon icon-fa-video-camera", 'aria-hidden':"true" });
+        }
+        else if(data.type == 'audio'){
+            icon = React.createElement('span',  { class:"icon icon-fa-music", 'aria-hidden':"true" });
+        }
```

Comment:
Draftail supports providing icons as SVG so we don’t need font awesome for these at least.